Fix:

error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
https://github.com/micans/mcl/commit/6ef00d56544232b9b6f04d979c4e3138df019471

error: implicit declaration of function 'mcxFree' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
https://github.com/micans/mcl/commit/7bfa704f99ecd43b4c5a7808ede9f131c7b5fc36

error: implicit declaration of function 'isatty' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
Fix is already present in initial commit:
https://github.com/micans/mcl/commit/32ce6dd2323d02cc79ae755f57a7d1fa0152221a
--- configure.orig	2014-05-16 16:59:55.000000000 -0500
+++ configure	2022-01-24 06:17:16.000000000 -0600
@@ -3725,6 +3725,7 @@
 
 #include <limits.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 unsigned test[] =
 {  0x00000000, 0x000000ff, 0x0000ff00, 0x00ff0000, 0xff000000
--- src/shcl/clmadjust.c.orig	2011-04-05 04:23:22.000000000 -0500
+++ src/shcl/clmadjust.c	2022-01-24 11:40:33.000000000 -0600
@@ -181,7 +181,7 @@
             ;  }
             }
          }
-         mcxFree(&cl)
+         mclxFree(&cl)
       ;  cl = cm
          /* mclxWrite(cm, xfout, MCLXIO_VALUE_NONE, RETURN_ON_FAIL) */
       ;  return 0
--- src/shmcx/mcxi.c.orig	2013-10-22 13:38:20.000000000 -0500
+++ src/shmcx/mcxi.c	2022-01-24 11:45:11.000000000 -0600
@@ -8,6 +8,7 @@
 */
 
 #include <stdio.h>
+#include <unistd.h>
 
 #include "stack.h"
 #include "glob.h"
